web/src/ts/store/auth/reducer.ts

```diff
@@ -0,0 +1,41 @@
+import { UserAction, AuthState, UserActionTypes } from '.'
+
+const initialState: AuthState = {
+  user: {
+    createdAt: null,
+    email: null,
+    id: null,
+    username: null,
+  },
+  loading: false,
+  isAuthenticated: false,
+  message: null,
+}
+
+export const authReducer = (state = initialState, action: UserAction): AuthState => {
+  switch (action.type) {
+    case UserActionTypes.REGISTRATION_USER:
```

Comment:
I believe REGISTRATION_USER is not a part of authentication. Only `sessions` endpoints are